packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,345 @@
+     *
+     * @link https://docs.aws.amazon.com/batch/latest/userguide/job_definition_parameters.html
+     * @default - undefined
+     */
+    readonly parameters?: { [key: string]: string };
+
+    /**
+     * The number of times to move a job to the RUNNABLE status. You may specify between 1 and
+     * 10 attempts. If the value of attempts is greater than one, the job is retried on failure
+     * the same number of attempts as the value.
+     *
+     * @default 1
+     */
+    readonly retryAttempts?: number;
+
+    /**
+     * The timeout configuration for jobs that are submitted with this job definition. You can specify
+     * a timeout duration after which AWS Batch terminates your jobs if they have not finished.
+     *
+     * @default - undefined
+     */
+    readonly timeout?: Duration;
+}
+
+/**
+ * Properties for specifying multi-node properties for compute resources
+ */
+export interface IMultiNodeProps {
+    /**
+     * Specifies the node index for the main node of a multi-node parallel job. This node index value must be fewer than the number of nodes.
+     */
+    mainNode: number;
+
+    /**
+     * A list of node ranges and their properties associated with a multi-node parallel job.
+     */
+    rangeProps: INodeRangeProps[];
+
+    /**
+     * The number of nodes associated with a multi-node parallel job.
+     */
+    count: number;
+}
+
+/**
+ * Properties for a multi-node batch job
+ */
+export interface INodeRangeProps {
+    /**
+     * The container details for the node range.
+     */
+    container: IJobDefinitionContainer;
+    /**
+     * The minimum node index value to apply this container definition against. If omitted, the value zero is used.
+     * You may nest node ranges, for example 0:10 and 4:5, in which case the 4:5 range properties override the 0:10 properties.
+     */
+    fromNodeIndex?: number;
+    /**
+     * The maximum node index value to apply this container definition against. If omitted, the highest value is used relative
+     * to the number of nodes associated with the job. You may nest node ranges, for example 0:10 and 4:5,
+     * in which case the 4:5 range properties override the 0:10 properties.
+     */
+    toNodeIndex?: number;
+}
+
+/**
+ * Properties of a job definition
+ */
+export interface IJobDefinition extends IResource {
+    /**
+     * The ARN of this batch job definition
+     *
+     * @attribute
+     * @default Cloudformation-generated ARN
+     */
+    readonly jobDefinitionArn: string;
+
+    /**
+     * The name of the batch job definition
+     *
+     * @attribute
+     * @default Cloudformation-generated name
+     */
+    readonly jobDefinitionName: string;
+}
+
+/**
+ * Batch Job Definition
+ *
+ * Defines a batch job definition to execute a specific batch job.
+ */
+export class JobDefinition extends Resource implements IJobDefinition {
+    /**
+     * Fetches an existing batch job definition by its amazon resource name.
```

Comment:
It actually doesn't fetch - it simply assumes that ARN exists. A better word would be "imports".

Reply:
Ohhhh, well hey, you learn something new everyday :P haha. Alright. I will reword it then to say:

`Assumes an existing ...`